### PR TITLE
Add EGA download tool (Pyega3)

### PR DESCRIPTION
**Pull request by shiltemann**
This adds a first implementation of a tool to download data from EGA, using the new python-based client (https://github.com/EGA-archive/ega-download-client)

Users can set their credentials under their user preferences (will PR to Galaxy to add the EGA credentials fields), and if unset, then the EGA test credentials are use (which has some 1000 genomes data)
 
There are enhancements to be made in the future, for example to list all the user's authorized files nicely in a dropdown orso, but for now this lets users obtain a list of their authorized datasets, list the files in a dataset, or download a file (or a specified genomic range of a file) 

FOR CONTRIBUTOR:
* [X] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [X] - License permits unrestricted use (educational + commercial)
* [X] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/pyega3/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# PyEGA3
+
+## set up user credentials on Galaxy
+
+To enable users to set their credentials for this tool,
+make sure the file `confgi/user_preferences_extra.yml` has the following section:
```
Comment (bernt-matthias) [inline tools/pyega3/README.md:6]: confgi -> config

File: tools/pyega3/pyega3.xml [hunk 2]
```diff
@@ -0,0 +1,137 @@
+<tool id="pyega3" name="EGA Download Client" version="@VERSION@+galaxy0" python_template_version="3.5">
```
Comment (bernt-matthias) [inline tools/pyega3/pyega3.xml:1]: Would if be a better option to use profile instead of python_template_version

File: tools/pyega3/pyega3.xml [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+        </data>
+        <data name="dataset_file_list" format="txt" label="${tool.name}: dataset file list">
+            <filter> action['action_type'] == 'list_dataset_files' </filter>
+        </data>
+        <data name="downloaded_file" auto_format="true" label="${tool.name}: ${action.file_id} ${action.range.reference_name} ${action.range.start} ${action.range.end}">
+            <filter> action['action_type'] == 'download_file' </filter>
+        </data>
+    </outputs>
+    <tests>
+        <test><!-- list datasets with default credentials -->
```
Comment (bernt-matthias) [inline tools/pyega3/pyega3.xml:90]: Can you add `expect_num_outputs="1"` to all tests?

File: tools/pyega3/pyega3.xml [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                    <validator type="regex" message="EGA dataset ID must be a string of numbers prefixed by 'EGAD'">EGAD[0-9]+</validator>
+                </param>
+            </when>
+            <when value="list_datasets"/>
+            <when value="download_file">
+                <param name="file_id" type="text" optional="false" label="EGA File Accession Identifier" help="Identifier starting with 'EGAF'. For example: EGAF00001753735">
+                     <validator type="regex" message="EGA Accession ID must be a string of numbers prefixed by 'EGAD' (datasets) or 'EGAF' (files)">EGAF[0-9]+</validator>
+                </param>
+                <section name="range" title="Request a specific Genomic range?" expanded="false">
+                <param argument="--reference-name" name="reference_name" type="text" optional="true" label="Reference Sequence Name" help="For example 'chr1', '1', or 'chrX'. If unspecified, all data is returned." />
```
Comment (bernt-matthias) [inline tools/pyega3/pyega3.xml:71]: `name` is not needed here. But in the cheetah code you should quote `'$action.range.reference_name'` unless you forbid spaces, or?

**Comment by bgruening** [comment]
ping @shiltemann ... if you address @bernt-matthias review we can merge this I think.

**Comment by shiltemann** [comment]
thanks @bernt-matthias, I made the updates you suggested

**Comment by shiltemann** [comment]
Failing test is a 500 error, might be a temporary issue on their side with retrieving genomic ranges,htsget orso (e.g. even if I revert the changes I made today, the previously passing test fails now). Will try again a bit later.

**Comment by shiltemann** [comment]
ok, updated the tool version, that worked for me locally at least to fix the test failure

**Review by bgruening — APPROVED** [review]

**Pull request merged**